cupy/linalg/_decomposition.py

```diff
@@ -459,9 +460,16 @@ def svd(a, full_matrices=True, compute_uv=True):
 
     buffersize = gesvd_bufferSize(handle, m, n)
     workspace = cupy.empty(buffersize, dtype=a_dtype)
+    if not runtime.is_hip:
+        # TODO(leofang): WHY????
+        rwork_ptr = 0
+    else:
+        rwork = cupy.empty(min(m, n)-1, dtype=s_dtype)
+        rwork_ptr = rwork.data.ptr
```

Comment:
On HIP if we don't allocate `rwork`, an error `rocblas_status_invalid_pointer` would be raised...

Reply:
What happens if you pass the same pointer to cuda?
This might be a bug in the cuda path

Reply:
Looks like passing `0` is allowed in cuSOLVER?
> The rwork is real array of dimension (min(m,n)-1). If devInfo>0 and rwork is not nil, rwork contains the unconverged superdiagonal elements of an upper bidiagonal matrix. This is slightly different from LAPACK which puts unconverged superdiagonal elements in work if type is real; in rwork if type is complex. rwork can be a NULL pointer if the user does not want the information from supperdiagonal.

https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverDN-lt-t-gt-gesvd

I didn't change the behavior on the CUDA side because it seems to work fine. Should I unify it?

Reply:
let's leave as is then :).
I saw some bugs in CuPy where the working space was not properly assigned before, so I thought this may be the case too. Thanks for the double check.

Reply:
However, some comment here wouldn't hurt

Reply:
On it.

Reply:
Done in d5af83f.